### PR TITLE
1. fixed error handling of getCompletedCourses

### PR DESCRIPTION
**Pull request by KevenLYF**
2. removed useless dependencies

### CODE REVIEW

File: backend/logic/users/users.js [hunk 1]
```diff
@@ -150,21 +174,23 @@ var getCompletedCourses = async studentID => {
...
   let results;
   try {
-    results = await connection.query(sql_query, [studentID]);
-    connection.release();
+    results = await conn.query(sql_query, [studentID]);
+    if (results.length !== 0) {
+      courses = JSON.stringify(results);
+      return courses;
+    }
+
```
Comment (zerfuth) [inline backend/logic/users/users.js]:
```suggestion
connection.release();
```
Still need to release connection, no?

Comment (zerfuth) [reply backend/logic/users/users.js]: NVM didn't see finally clause

**Comment by KevenLYF** [comment]
@zerfuth  thanks, yeah it will always be executed no matter what you are doing before

File: backend/logic/users/users.js [hunk 2]
```diff
@@ -150,21 +174,23 @@ var getCompletedCourses = async studentID => {
...
+    return "No student ID found!";
```
Comment (AlexLam1997) [inline backend/logic/users/users.js]: Its also possible that the student simply doesnt have any completed courses, I would change this to initialise courses to be an empty array, update it if (results.length!=0) or else return empty array 
```
let courses = []
if (results.length !== 0) {
      courses = JSON.stringify(results);
    }
return courses;

```

**Comment by KevenLYF** [comment]
@AlexLam1997  Actually I thought about it too, but the table `student_course_offering` is meant to record the courses that a student has completed, so it cannot be empty.

**Comment by AlexLam1997** [comment]
> @AlexLam1997 Actually I thought about it too, but the table `student_course_offering` is meant to record the courses that a student has completed, so it cannot be empty.

Hm but what if the student is newly registered in our service and has not yet imputted his list of completed courses? Would it not be empty then?

**Pull request merged**